src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpThurloeDAO.scala

```diff
@@ -119,6 +118,33 @@ class HttpThurloeDAO ( implicit val system: ActorSystem, implicit val executionC
     }
   }
 
+  override def bulkUserQuery(userIds: List[String], keySelection: List[String]): Future[List[ProfileWrapper]] = {
+    val userIdParams:List[(String,String)] = userIds.map(("userId", _))
+    val keyParams:List[(String,String)] = keySelection.map(("key", _))
+
+    val allQueryParams = userIdParams ++ keyParams
+
+    val queryUri = Uri(UserApiService.remoteGetQueryURL).withQuery(allQueryParams:_*)
+
+    // default uri length for Spray - which Thurloe uses - is 2048 chars
+    assert(queryUri.toString().length <  2048, s"generated url is too long at ${queryUri.toString().length} chars.")
```

Comment:
Instead of `assert` which throws an `AssertionError` and tends to crash the ActorSystem in my experience -- should this return a `Future.failed(...)`?